CIScripts/Test/TestConfigurationUtils.Tests.ps1

```diff
@@ -0,0 +1,48 @@
+
+. $PSScriptRoot\TestConfigurationUtils.ps1
+
+Describe "Select-ValidNetIPInterface unit tests" -Tags CI, Unit {
+        It "Both AddressFamily and SuffixOrigin match values in Select-ValidNetIPInterface" {
+            $TestResult = $MockedGetNetIPAddress | Select-ValidNetIPInterface
+            $TestResult | Should -eq $MockedGetNetIPAddress
+        }
+
+        It "AddressFamily isn't matching" {
+            $MockedGetNetIPAddress.AddressFamily = "IPv6"
+            $TestResult = $MockedGetNetIPAddress | Select-ValidNetIPInterface
+            $TestResult | Should BeNullOrEmpty
+        }
+
+        It "SuffixOrigin isn't matching" {
+            $MockedGetNetIPAddress.SuffixOrigin = "WellKnown", "Link", "Random"
```

Comment:
Is this `$MockedGetNetIPAddress` a copy of `$MockedGetNetIPAddress` defined in `BeforeEach` or is it actually a global variable?

Anyway, it would be better to avoid sharing global state between tests and construct a fresh object each time.

Reply:
Sorry for the comment above, I was somehow thinking you're doing `BeforeAll`, not `BeforeEach`.